avm/res/network/p2s-vpn-gateway/tests/e2e/defaults/main.test.bicep

```diff
@@ -0,0 +1,66 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.p2svpngateway-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'vscmin'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    virtualWANName: 'dep-${namePrefix}-vw-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
```

Comment:
Please note that the following parameters are defined even though they're optional (and 'should' hence not be defined in the defaults test):
```
associatedRouteTableName: 'defaultRouteTable'
location: resourceLocation
vpnClientAddressPoolAddressPrefixes: [
  '10.0.2.0/24'
]
```

Reply:
For the defaults test; are we talking:

- the minimum configuration to get the resource to actually deploy
- or, the minimum configuration to deploy _and_ have it actually be functional?

Reply:
😄 

The ideally both? the minimum test should ensure that the defaults of the module work (hence the name) which usually should still mean the resource works. If this is not possible for this resource type because e.g., there are different deployment variants that require different parameters to be provided maybe deploying one variant is the better idea.
At the end of the day the idea should just be to make sure the module works the way it is intended. So let's take a closer look at the 3 highlighted parameters
- `associatedRouteTableName`: I guess this could be a module default, if required? Unless this breaks something else.
- `location`: Likely not required as the module's default should anyways be the one provided
- `vpnClientAddressPoolAddressPrefixes`: The one that may have the biggest impact on functionality, I guess? And hence maybe the one to keep around.